server/src/main/java/org/opensearch/index/engine/exec/merge/OneMerge.java

```diff
@@ -23,6 +24,26 @@ public List<CatalogSnapshot.Segment> getSegmentsToMerge() {
         return segmentsToMerge;
     }
 
+    public long getTotalSizeInBytes() {
+        long totalSize = 0;
+        for (CatalogSnapshot.Segment segment : segmentsToMerge) {
+            for (WriterFileSet writerFileSet : segment.getDFGroupedSearchableFiles().values()) {
+                totalSize += writerFileSet.getTotalSize();
+            }
+        }
+        return totalSize;
+    }
+
+    public long getTotalNumDocs() {
+        long totalDocs = 0;
+        for (CatalogSnapshot.Segment segment : segmentsToMerge) {
+            for (WriterFileSet writerFileSet : segment.getDFGroupedSearchableFiles().values()) {
+                totalDocs += writerFileSet.getNumRows();
+            }
+        }
+        return totalDocs;
+    }
+
```

Comment:
Instead of calculating it everytime, we can calculate this during oneMerge creation and store.